ding_periodical.module

```diff
@@ -135,12 +136,17 @@ function theme_ding_periodical_issues($variables) {
 
       // Check if the volume is reservable and add reservation button.
       if (!empty($issue_id) && ding_periodical_is_reservable($issues[$key])) {
-        $path = array($issue_id, $volume, $key);
+        $item_id = array($issue_id, $volume, $key);
+
+        foreach ($item_id as &$v) {
+          $v = rawurlencode($v);
```

Comment:
rawurlencode vs base64_encode are used here for two different purposes:
rawurlencode - encodes the id so the lms system accepts the id properly
e.g.
'Vol. 40, No. 459 &amp; 460' - not encoded, doesn't reserve, error ReservationNoHolding
'Vol.%2040%2C%20No.%20459%20%26%20460' - encoded, does reserve

base64_encode - eliminates special characters so it can pass thru the hook_menu